sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -429,14 +429,16 @@ private SpecializedBlobClientBuilder prepareBuilder() {
             Function<Flux<ByteBuffer>, Mono<Response<BlockBlobItem>>> uploadInChunksFunction = (stream) ->
                 uploadInChunks(blockBlobAsyncClient, stream, validatedParallelTransferOptions,
                     options.getHeaders(), options.getMetadata(), options.getTags(),
-                    options.getTier(), validatedRequestConditions);
+                    options.getTier(), validatedRequestConditions, options.isComputeMd5());
 
             BiFunction<Flux<ByteBuffer>, Long, Mono<Response<BlockBlobItem>>> uploadFullBlobMethod =
-                (stream, length) -> blockBlobAsyncClient.uploadWithResponse(new BlockBlobSimpleUploadOptions(
-                    ProgressReporter.addProgressReporting(stream,
-                        validatedParallelTransferOptions.getProgressReceiver()), length)
+                (stream, length) -> UploadUtils.computeMd5(ProgressReporter.addProgressReporting(stream,
```

Comment:
This is the only part of the code that I'm iffy about. Like wrapping the ProgressReporter and stuff looks a little unclean. I could just make a method maybe and make it look a little better.